static_compress/mixin.py

```diff
@@ -71,47 +72,64 @@ def get_modified_time(self, name):
         return self._datetime_from_timestamp(getmtime(alt))
 
     def post_process(self, paths, dry_run=False, **options):
-        if hasattr(super(), "post_process"):
-            yield from super().post_process(paths, dry_run, **options)
-
         if dry_run:
             return
 
+        files = OrderedDict()
+
+        super_class = super()
+        if hasattr(super_class, "post_process"):
+            generator = super_class.post_process(paths, dry_run, **options)
+        else:
+            generator = iter((f, f, False) for f in paths)
+
+        for file, hashed_file, processed in generator:
+            files[file] = hashed_file
```

Comment:
It's not exactly clear what this is doing. Consider adding commenting.

I'm guessing that it is to not process files that are not emitted by super class. If possible please consider splitting this to another PR so we can review this easier.

Reply:
not sure here.
a super class must not have a post_process() method implemented. so you have to take in this case the whole fileset, that have not been post_processed